emcc.py

```diff
@@ -3150,6 +3150,8 @@ def process_libraries(libs, lib_dirs, temp_files):
       if jslibs is not None:
         libraries += [(i, jslib) for jslib in jslibs]
         consumed.append(i)
+      elif building.map_and_apply_to_settings(lib):
```

Comment:
Should this just be `map_to_settings` ?   I'm not sure either is that descriptive so I'm fine either way.  

I guess if it was called `map_to_settings` then it would return the new settings and you would apply it here?

Reply:
I considered that, but then returning a list of key mappings and applying them seems odd. a pretty complex interface, compared to just applying them there.